src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -158,6 +161,31 @@ internal static bool VerifyRestoreOutput(RestoreRequest request, CacheFile cache
                 return false;
             }
 
+            if (request.ForceUpdatePackageLastAccessTime)
+            {
+                foreach (var package in cacheFile.ExpectedPackageFilePaths)
+                {
+                    var packageFolder = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Parent.FullName;
+
+                    // we avoid fallback folders as they are readonly
+                    if (request.Project.RestoreMetadata.FallbackFolders.Any(fallbackFolder => fallbackFolder.Equals(packageFolder, StringComparison.OrdinalIgnoreCase)))
+                        continue;
+
+                    var packageRoot = Directory.GetParent(package).FullName;
+                    var metadataFile = Path.Combine(packageRoot, ".nupkg.metadata");
```

Comment:
There *might* be a method that can do this.

Reply:
Pushed a version that uses `VersionFolderPathResolver`.